lib/probe/http-client.js

```diff
@@ -247,10 +247,19 @@ class HttpRequestMetrics {
         // ...otherwise count actual response size as its streamed
         if (isNaN(this.responseSize)) {
             this.responseSize = 0;
+
+            // prevent stream from being read as soon as we add our 'data' listener below,
+            // leave that to the actual client code consuming the request
+            response.readableFlowing = false;
+
             response.on("data", (chunk) => {
                 debug("⚡️ [response] data", chunk);
                 this.responseSize += chunk.length;
             });
+
+            // reset flowing state so that as soon as client code adds a data listener
+            // it will start consuming the stream
+            response.readableFlowing = null;
```

Comment:
I didn't understand the reason for setting `readableFlowing` to `null`. Docs says:

> Calling readable.pause(), readable.unpipe(), or receiving backpressure will cause the readable.readableFlowing to be set as false, temporarily halting the flowing of events but not halting the generation of data. While in this state, attaching a listener for the 'data' event will not switch readable.readableFlowing to true.


Besides, what happens if the client starts consuming before instrumentation sets `readableFlowing` to `null`? Wouldn't it cause the client to hang?

Reply:
If you don't set it back to `null`, the stream will not start when the actual client code registers their `data` listener.